intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/documentation/XtextCalleeDocumentationProvider.xtend

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.idea.documentation
+
+import com.google.inject.Inject
+import org.eclipse.xtext.documentation.IEObjectDocumentationProvider
+import org.eclipse.xtext.idea.filesystem.IdeaWorkspaceConfigProvider
+import org.eclipse.xtext.naming.IQualifiedNameProvider
+import org.eclipse.xtext.psi.PsiEObject
+
+import static extension org.eclipse.xtext.EcoreUtil2.*
+
+/**
+ * @author Moritz Eysholdt - Initial contribution and API
+ */
+class XtextCalleeDocumentationProvider {
+
+	@Inject IdeaWorkspaceConfigProvider workspaceCfgProvider
+	@Inject IQualifiedNameProvider qNameProvider
+	@Inject IEObjectDocumentationProvider eObjectDocProvider
+
+	def protected getElementInfo(PsiEObject element) {
+		val obj = element.EObject
+		val name = qNameProvider.getFullyQualifiedName(obj)
+		if (name !== null) {
+			return '''<b>�obj.eClass.name� '�name�'</b>'''
+		} else {
+			val container = obj.allContainers.map[qNameProvider.getFullyQualifiedName(it)].findFirst[it !== null]
+			return '''<b>�obj.eClass.name� in '�container�'</b>'''
+		}
+	}
+
+	def protected getFileInfo(PsiEObject element) {
+		val resource = element.EObject.eResource
+		val module = workspaceCfgProvider.getWorkspaceConfig(resource.resourceSet).findProjectContaining(resource.URI)
+		val uri = resource.URI.deresolve(module.path)
```

Comment:
This will display an absolute URI if the resource is not physically contained in the module folder. I guess we should rather deresolve against the containing source folder and include the source folders name in the file info.